internal/snykclient/sbomconvert.go

```diff
@@ -0,0 +1,79 @@
+package snykclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/snyk/cli-extension-sbom/internal/errors"
+)
+
+const (
+	sbomConvertAPIVersion = "2025-03-06~beta"
+	MIMETypeOctetStream   = "application/octet-stream"
+)
+
+func (t *SnykClient) SBOMConvert(
+	ctx context.Context,
+	errFactory *errors.ErrorFactory,
+	sbom io.Reader,
+) ([]ScanResult, error) {
+	u, err := buildAPIURL(t.apiBaseURL, sbomConvertAPIVersion, t.orgID)
+	if err != nil {
+		return nil, fmt.Errorf("sbom convert api url invalid: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		u,
+		sbom,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set(ContentTypeHeader, MIMETypeOctetStream)
+
+	resp, err := t.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return nil, newStatusCodeError(resp.StatusCode)
+	}
+
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response: %w", err)
+	}
+
+	var convertResp SBOMConvertResponse
+	err = json.Unmarshal(bodyBytes, &convertResp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal JSON response: %w", err)
+	}
+
+	return convertResp.ScanResults, nil
+}
+
+func buildAPIURL(apiBaseURL, apiVersion, orgID string) (string, error) {
+	base, err := url.Parse(apiBaseURL)
+	if err != nil {
+		return "", err
+	}
+
+	base.Path = path.Join(base.Path, "hidden/orgs", orgID, "sboms/convert")
```

Comment:
would this not result in backslashes on windows platforms? better sprintf with `"/"`?

Reply:
Don't we actually want this?

```suggestion
	base.Path = path.Join(base.Path, "hidden", "orgs", orgID, "sboms", "convert")
```